web/client/epics/geostories.js

```diff
@@ -0,0 +1,83 @@
+const { searchParamsSelector, searchTextSelector, totalCountSelector} = require('../selectors/geostories');
+const GeoStoreApi = require('../api/GeoStoreDAO');
+const { wrapStartStop } = require('../observables/epics');
+const {error} = require('../actions/notifications');
+
+const {deleteResource} = require('../api/persistence');
+
+const calculateNewParams = state => {
+    const totalCount = totalCountSelector(state);
+    const {start, limit, ...params} = searchParamsSelector(state) || {};
+    if (start === totalCount - 1) {
+        return {
+            start: Math.max(0, start - limit),
+            limit
+        };
+    }
+    return {
+        start, limit, ...params
+    };
+};
+
+module.exports = {
+    searchGeostoriesOnMapSearch: action$ =>
+        action$.ofType(MAPS_LIST_LOADING)
+            .switchMap(({ searchText }) => Rx.Observable.of(searchGeostories(searchText))),
+    searchGeostories: (action$, { getState = () => { } }) =>
+        action$.ofType(SEARCH_GEOSTORIES)
+            .map( ({params, searchText, geoStoreUrl}) => ({
+                searchText,
+                options: {
+                    params: params || searchParamsSelector(getState()) || {start: 0, limit: 12},
+                    ...(geoStoreUrl ? { baseURL: geoStoreUrl } : {})
+                }
+            }))
+            .switchMap(
+                ({ searchText, options }) =>
+                    Rx.Observable.defer(() => GeoStoreApi.getResourcesByCategory("GEOSTORY", searchText, options))
+                        .map(results => geostoriesListLoaded(results, {searchText, options}))
+                        .let(wrapStartStop(
+                            geostoriesLoading(true, "loading"),
+                            geostoriesLoading(false, "loading"),
+                            () => Rx.Observable.of(error({
+                                title: "notification.error",
+                                message: "resources.geostories.errorLoadingGeostories",
+                                autoDismiss: 6,
+                                position: "tc"
+                            }))
+                        ))
+            ),
+    deleteDashboard: action$ => action$
+        .ofType(DELETE_GEOSTORY)
+        .switchMap(id => deleteResource(id).map(() => geostoryDeleted(id)))
+        .let(wrapStartStop(
+            geostoriesLoading(true, "loading"),
+            geostoriesLoading(false, "loading"),
+            () => Rx.Observable.of(error({
+                title: "notification.error",
+                message: "resources.geostories.deleteError",
+                autoDismiss: 6,
+                position: "tc"
+            }))
+        )),
+    reloadOnGeostories: (action$, { getState = () => { } }) =>
+        action$.ofType(GEOSTORY_DELETED, MAP_DELETED, MAP_METADATA_UPDATED, RELOAD, ATTRIBUTE_UPDATED, GEOSTORY_SAVED)
```

Comment:
i thin this has too much stuff (MAP_DELETED etc), is it ?

Reply:
I'll check it